src/app/routes/broker-ui/catalog-page/catalog-page/catalog-page.component.ts

```diff
@@ -50,13 +52,16 @@ export class CatalogPageComponent implements OnInit, OnDestroy {
     private assetDetailDialogService: AssetDetailDialogService,
     private brokerServerApiService: BrokerServerApiService,
     private store: Store,
+    private route: ActivatedRoute,
+    private router: Router,
   ) {}
 
   ngOnInit(): void {
     this.store.dispatch(CatalogPage.Reset);
     this.startListeningToStore();
     this.startEmittingSearchText();
     this.startEmittingSortBy();
+    this.parseConnectorEndpointQueryParams();
```

Comment:
I tried this but the problem with this approach is that we have to wait for the first fetch to be Ready because it returns the available filters, which can then be set